fastlane/lib/fastlane/fastlane_require.rb

```diff
@@ -39,19 +39,22 @@ def install_gem_if_needed(gem_name: nil, require_gem: true)
       end
 
       def gem_installed?(name, req = Gem::Requirement.default)
-        # We fork and try to load a gem with that name in the child process so we
-        # don't actually load anything we don't want to load
-        # This is just to test if the gem is already preinstalled, e.g. YAML
+        # In special cases a gem is already preinstalled, e.g. YAML.
+        # To find out we try to load a gem with that name in a child process
+        # (so we don't actually load anything we don't want to load)
         # See https://github.com/fastlane/fastlane/issues/6951
-        fork do
-          begin
-            require name
-          rescue LoadError
-            exit(1)
+        if Process.respond_to?(:fork)
```

Comment:
You could instead do something like the following:

```ruby
require_tester = <<-RB.gsub(/^ */, '')
  begin
    require ARGV.first
  rescue LoadError
    exit(1)
  end
RB
system(RbConfig.ruby, "-e", require_tester.lines.map(&:chomp).join("; "), name)
return true if $?.success?
```

Which should work on all of Linux, Windows and macOS

Reply:
Ah, that's what you meant in that comment of the commit where this was added ;)

Without having tried it, this looks like it could do what it should do.

I am still unsure how to actually test this... I am really hesitant on mess with this (hopefully working) code without knowing how I can replicate this situation or best even write a test that makes sure this code does what it should do. Ideas?

Reply:
Sure, the whole purpose of this code was just to test whether a module comes preinstalled, like `time` or `yaml`. If you run `Fastlane::FastlaneRequire.gem_installed?("yaml")` it return `true` if this code works and `false` if it doesn't.

The reason this was added was that the line below `Gem::Specification.any? #...` returns true *only* for gems that were installed with RubyGems, not anything else that would still be a valid `require` statement. 

For example, fire up `bundle exec pry` and try out the following statements:

```ruby
[1] pry(main)> Gem::Specification.any { |s| s.name == "yaml" }
=> false
[2] pry(main)> require "yaml"
=> true
[3] pry(main)> require "blabla"
LoadError: cannot load such file -- blabla
from (pry):2:in `require'
[4] pry(main)> require_relative 'fastlane/lib/fastlane/fastlane_require'
=> true
[5] pry(main)> Fastlane::FastlaneRequire.gem_installed?("yaml")
=> true
[6] pry(main)> Fastlane::FastlaneRequire.gem_installed?("blabla")
=> false
```

Reply:
Ah ok.
On which systems a module is preinstalled? Is this default for all ruby installations?

Reply:
There's a couple that come default for all ruby installations ... why they don't get required I have no idea however. I don't know if there's a list anywhere either

Reply:
Thanks for the explanation, I finally understood what is happening here and wrote some tests for the method that show it working on Ubuntu and macOS (and failing locally on Windows). Now I can continue to work on the code and try what you posted above.

Reply:
Done. I also moved the "normal" case with the simple check before the special, edge case handling. 

Works perfectly :)